lightning/src/ln/peer_handler.rs

```diff
@@ -864,7 +869,7 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: Deref, CMH: Deref> P
 									peer.their_node_id = Some(their_node_id);
 									insert_node_id!();
 									let features = InitFeatures::known();
-									let resp = msgs::Init { features };
+									let resp = msgs::Init { features, remote_network_address: peer.their_net_address.clone() };
```

Comment:
We should filter out local and private addresses (ie only include v6 if its in 2000::/3, skip v4 of 0/8, 10/8, 100.64/10, 127/8, 169.254/16, 172.16/12, 192.88.99/24, 192.168/16, etc).

Reply:
> This should describe when a user should pass a remote_network_address, what the reason is, and what considerations they should think about when deciding whether or not to include it.

I found this on the original issue
```
This adds the option to report an remote IP address back to a connecting
peer using the init message. A node can decide to use that information
to discover a potential update to its public IPv4 address (NAT) and use
that for a node_announcement update message containg the new address.
```
I am not sure when the user should pass it and what are the considerations

Reply:
Basically, users should pass in whatever the host is that they connected to.

Reply:
> We should filter out local and private addresses (ie only include v6 if its in 2000::/3, skip v4 of 0/8, 10/8, 100.64/10, 127/8, 169.254/16, 172.16/12, 192.88.99/24, 192.168/16, etc).

How can I filter these? The `NetAddress` doesn't have a subnet mask field

Reply:
You'll have to implement masking :)

Reply:
Should I add a `mask` field in the `NetAddress` enum?

Reply:
Oh, no, we need to check if the ip in `NetAddress` is in the above netblocks, not add a mask.

Reply:
Okay, I will try to implement this.

Reply:
> We should filter out local and private addresses (ie only include v6 if its in 2000::/3, skip v4 of 0/8, 10/8, 100.64/10, 127/8, 169.254/16, 172.16/12, 192.88.99/24, 192.168/16, etc).

I was able to get all the ranges of ipv4 addresses but I am not able to convert the ipv6 address(2000::/3) to a 16 byte array

Reply:
What did you try to convert it?

Reply:
I am using `match` to check if the `peer.their_net_address.clone()` matches with the reserved IP addresses. In IPv4, it is easier to find the range which is to be excluded( like from 0.0.0.0 to 0.255.255.255 for 0/8). But in ipv6, the address is of the form `2000:0000:0000:0000:0000:0000:0000:0000` but in the `NetAddress`, the ipv6 address should be a 16-byte array

Reply:
Right, did you try converting the hex into bytes :)